lib/identity_cache.rb

```diff
@@ -202,42 +202,46 @@ def fetch_multi(*keys)
       result
     end
 
-    # Executes a block with deferred parent expiration, ensuring that the parent
-    # records' cache expiration is deferred until the block completes. When the block
-    # completes, it triggers expiration of the primary index for the parent records.
-    # Raises a NestedDeferredParentBlockError if a deferred parent expiration block
-    # is already active on the current thread.
+    # Executes a block with deferred cache expiration, ensuring that the records' (parent,
+    # children and attributes) cache expiration is deferred until the block completes. When
+    # the block completes, it issues delete_multi calls for all the records and attributes
+    # that were marked for expiration.
     #
     # == Parameters:
     # No parameters.
     #
     # == Raises:
-    # NestedDeferredParentBlockError if a deferred parent expiration block is already active.
+    # NestedDeferredCacheExpirationBlockError if a deferred cache expiration block is already active.
     #
     # == Yield:
-    # Runs the provided block with deferred parent expiration.
+    # Runs the provided block with deferred cache expiration.
     #
     # == Returns:
     # The result of executing the provided block.
     #
     # == Ensures:
-    # Cleans up thread-local variables related to deferred parent expiration regardless
+    # Cleans up thread-local variables related to deferred cache expiration regardless
     # of whether the block raises an exception.
-    def with_deferred_parent_expiration
-      raise NestedDeferredParentBlockError if Thread.current[:idc_deferred_parent_expiration]
+    def with_deferred_expiration
+      raise NestedDeferredCacheExpirationBlockError if Thread.current[:idc_deferred_expiration]
 
-      Thread.current[:idc_deferred_parent_expiration] = true
-      Thread.current[:idc_parent_records_for_cache_expiry] = Set.new
+      Thread.current[:idc_deferred_expiration] = true
+      Thread.current[:idc_parent_records_to_expire] = Set.new
+      Thread.current[:idc_child_records_to_expire] = Set.new
+      Thread.current[:idc_attributes_to_expire] = Set.new
```

Comment:
Should we just use one `Set` for all the three types of keys: `parent`, `child`, and `attributes`? We just delete them all in one `write_multi` call.